pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -377,7 +489,13 @@ def supports_derivatives(
             Bool: Whether or not a derivative can be calculated provided the given information
 
         """
-        return 0
+        if execution_config is None and circuit is None:
+            return True
+        if execution_config.gradient_method not in {"adjoint", "best"}:
+            return False
+        if circuit is None:
+            return True
```

Comment:
```suggestion
        if circuit is None or (execution_config is None and circuit is None):
            return True
        if execution_config.gradient_method not in {"adjoint", "best"}:
            return False
```

Reply:
I try this change but break something in the unit test. :sweat_smile:  
The explanation is the following:
> if `circuit` is None but `execution_config` != None then the logics need to reach the last `return`. With this suggestion, the return always will be `True` and never reach the last `return`